src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,20 +2,96 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
 
-import static org.junit.Assert.assertTrue;
+import java.math.BigDecimal;
+import java.util.NoSuchElementException;
 
-public class ArrayListProductDaoTest
-{
-    private ProductDao productDao;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+public class ArrayListProductDaoTest {
+    private ArrayListProductDao productDao;
+
+    @Mock
+    private Product product1;
+    @Mock
+    private Product product2;
+    @Mock
+    private Product product3;
 
     @Before
     public void setup() {
+        MockitoAnnotations.initMocks(this);
+
         productDao = new ArrayListProductDao();
+        productDao.getProductList().add(product1);
+
+        when(product1.getId()).thenReturn(1L);
+        when(product1.getPrice()).thenReturn(new BigDecimal(100));
+        when(product1.getStock()).thenReturn(100);
+
+        when(product3.getId()).thenReturn(3L);
+        when(product3.getPrice()).thenReturn(new BigDecimal(300));
+        when(product3.getStock()).thenReturn(300);
     }
 
     @Test
-    public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+    public void testGetProduct() {
+        product2 = productDao.getProduct(1L);
```

Comment:
Wat.

Reply:
You've got product1 saved in your database. product1.getId() returns 1L -> product dao should return product1 when getProduct(1L) is called. And that's what needs to be checked.